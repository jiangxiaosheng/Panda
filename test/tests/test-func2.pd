func foo(a: int, b: float, c: string, d: boolean): void {

	for var i = 0; i < a; i + 1 {
		print("" + b)
		if d {
			print(c)
		}
	}

}

func main() {
    foo(10, 10.1, "hello", true)
}