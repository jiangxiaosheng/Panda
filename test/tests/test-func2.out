Sementically checked program: 


void foo(a, b, c, d) {
for (var i: int(int : 0)
; (bool : (int : i) < (int : a)); (int : (int : i) + (int : 1))) {
(int : print((string : (string : ) + (string : b))))
if ((bool : d)) {
(int : print((string : c)))
}
}
}

void main() {
(void : foo((int : 10), (float : 10.1), (string : hello), (bool : true)))
}

